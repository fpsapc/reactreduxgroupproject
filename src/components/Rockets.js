import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { reserveRocket } from '../redux/rockets/rocketsSlice';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  const dispatch = useDispatch();
  const handleReserveClick = (id) => {
    dispatch(reserveRocket({ id }));
  };

  return (
    <div>
      {rockets.map((rocket) => (
        <div className="rocketContainer" key={rocket.id}>
          <div>
            <img src={rocket.flickrImage[0]} alt="Rocket" style={{ width: '15rem' }} />
          </div>
          <div className="rocketData">
            <h2>{rocket.rocketName}</h2>
            <p>{rocket.rocketType}</p>
            <p>
              <span className="reservationBadge">
                {' '}
                {rocket.reserved ? 'Reserved' : ' ' }
                {' '}
              </span>
              {' '}
              {rocket.description}
            </p>
            <button type="button" className={rocket.reserved ? 'reservedBtn' : 'reserveBtn'} onClick={() => handleReserveClick(rocket.id)}>{rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}</button>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Rockets;
