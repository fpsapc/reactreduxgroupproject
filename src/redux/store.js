import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import dragonReducer, { fetchDragons } from './dragons/dragonSlice';
import rocketReducer, { fetchRockets } from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    dragons: dragonReducer,
    missions: missionReducer,
  },
});

store.dispatch(fetchRockets());
store.dispatch(fetchDragons());

export default store;
