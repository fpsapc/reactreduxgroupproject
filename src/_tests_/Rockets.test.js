import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Rockets from '../components/Rockets';

test('render correctly and matches the Rockets snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
