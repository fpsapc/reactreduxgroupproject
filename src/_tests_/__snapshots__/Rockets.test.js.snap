// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly and matches the Rockets snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
